app/code/core/Mage/Sales/Model/Quote.php

```diff
@@ -2089,4 +2087,21 @@ public function save()
         }
         return parent::save();
     }
+
+    /**
+     * @return string
+     */
+    public function getCouponCode(): string
+    {
+        return (string)$this->_getData('coupon_code');
+    }
+
+    /**
+     * @param string $couponCode
+     * @return $this
+     */
+    public function setCouponCode(string $couponCode) #static with php74
```

Comment:
what does "static with php74" comment mean?

Reply:
When adding new methods i tried to add type hints and return types if possible ... , was not sure about adding `self`.  Its a php8 change not 7.4 (https://php.watch/versions/8.0/static-return-type) ... comment can be removed.

However ... this should not be merged yet. If it's okay to add some getter/setter to avoid changes in templates, i'd also fix the other issues found by @elidrissidev.